### PR TITLE
JDK-8309964: Use directed inheritDoc for javax.lang.model API

### PR DESCRIPTION
**Pull request by jddarcy**
Update javax.lang.model and javax.annotation.processing to use directed inheritDoc.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8310172](https://bugs.openjdk.org/browse/JDK-8310172) to be approved

### Issues
 * [JDK-8309964](https://bugs.openjdk.org/browse/JDK-8309964): Use directed inheritDoc for javax.lang.model API (**Enhancement** - P4)
 * [JDK-8310172](https://bugs.openjdk.org/browse/JDK-8310172): Use directed inheritDoc for javax.lang.model API (**CSR**)

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14477/head:pull/14477` \
`$ git checkout pull/14477`

Update a local copy of the PR: \
`$ git checkout pull/14477` \
`$ git pull https://git.openjdk.org/jdk.git pull/14477/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14477`

View PR using the GUI difftool: \
`$ git pr show -t 14477`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14477.diff">https://git.openjdk.org/jdk/pull/14477.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14477#issuecomment-1593801145)

### CODE REVIEW

**Comment by jddarcy** [comment]
The bulk of this change is a refactoring of the inheritDoc tags to be better structured. The user-visible differences are detailed in the specdiff of the CSR; in summary:

-     Improvements to general spec vs implSpec for AbstractProcessor
-     Minor update to AbstractTypeVisitor8
-     Add explicit visitRecordComponent override to ElementScanner6; this changes where ElementScanner{7, 8, 9, 14} inherit the method from
-     SimpleTypeVisitor6 remove redundant text that was previous moved to an implSpec tag.

File: src/java.compiler/share/classes/javax/lang/model/util/AbstractTypeVisitor6.java [hunk 1]
```diff
@@ -138,15 +139,15 @@ public R visitIntersection(IntersectionType t, P p) {
...
      * AbstractTypeVisitor6} will always throw {@code
      * new UnknownTypeException(t, p)}.  This behavior is not required of a
      * subclass.
      *
-     * @param t  {@inheritDoc}
-     * @param p  {@inheritDoc}
+     * @param t  {@inheritDoc TypeVisitor}
+     * @param p  {@inheritDoc TypeVisitor}
      * @return a visitor-specified result
      * @throws UnknownTypeException
```
Comment (pavelrappo) [inline src/java.compiler/share/classes/javax/lang/model/util/AbstractTypeVisitor6.java:152]: Similar comment about inheriting exception info.

Comment (jddarcy) [reply src/java.compiler/share/classes/javax/lang/model/util/AbstractTypeVisitor6.java]: Good idea; I included the annotation value variant too -- I'll update the CSR for the change in annotation value visitor class.

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

**Pull request closed without merging**